### PR TITLE
Make build-images workflow name dynamic

### PR DESCRIPTION
**Pull request by potiuk**
As of today you can dynamically name your build-images workflow. This will be nice change to be able to distinguish the "tests" from build images workflows.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
More info here: https://github.blog/changelog/2022-09-26-github-actions-dynamic-names-for-workflow-runs/

**Comment by potiuk** [comment]
Screenshot 2022-09-27 at 03 30 04

File: .github/workflows/build-images.yml [hunk 1]
```diff
@@ -17,6 +17,7 @@
 #
 ---
 name: "Build Images"
+run-name: Build images for ${{ github.event.pull_request.title }}
```
Comment (uranusjr) [inline .github/workflows/build-images.yml]: Would it be a good idea to also include the PR number?

Comment (potiuk) [reply .github/workflows/build-images.yml]: Possibly even URL to PR. I wonder how it will look like.

**Comment by potiuk** [comment]
Maybe not beutiful, but you rarely look at this workflow and having a direct link you can copy&paste is pretty useful  


Screenshot 2022-09-27 at 10 18 43

**Comment by potiuk** [comment]
Should be good to go @uranusjr

**Review by uranusjr — APPROVED** [review]

**Comment by jedcunningham** [comment]
Looking forward to this, thanks!

**Pull request merged**